app/src/main/java/com/example/househomey/AddItemFragment.java

```diff
@@ -51,39 +60,82 @@ public AddItemFragment(CollectionReference itemRef) {
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View rootView = inflater.inflate(R.layout.fragment_add_item, container, false);
+        dateTextView = rootView.findViewById(R.id.add_item_date);
+
+        // Create DatePicker with dates restricted to past/present
+        CalendarConstraints.Builder constraintsBuilder = new CalendarConstraints.Builder();
+        constraintsBuilder.setEnd(System.currentTimeMillis());
+        constraintsBuilder.setValidator(DateValidatorPointBackward.now());
+        MaterialDatePicker<Long> datePicker = MaterialDatePicker.Builder.datePicker()
+                .setInputMode(MaterialDatePicker.INPUT_MODE_CALENDAR)
+                .setCalendarConstraints(constraintsBuilder.build())
+                .build();
+        datePicker.addOnPositiveButtonClickListener(selection -> {
+            dateAcquired = new Date(selection);
+            dateTextView.setText(datePicker.getHeaderText());
+            ((TextInputLayout) getView().findViewById(R.id.add_item_date_layout)).setError(null);
+        });
 
         // Show DatePicker when date field selected
-        dateTextView = rootView.findViewById(R.id.add_item_date);
-        dateTextView.setOnClickListener(v -> showDatePicker());
```

Comment:
Why did we get rid of this? If there's anyway to consolidate all of this date picker stuff into a method that would make this much cleaner. Maybe lines 63 -> 103 in a function

Reply:
So I realized that in that previous version, I was re-building the DatePicker every single time the show method was called. When really, it only needs to be built once upon Fragment init. Obviously this was bad for performance.

I can definitely consolidate the build part into an initialization function though, if you would like it to be cleaner. Something like `buildDatePicker`

Reply:
Yeah if we can have build and show that would be great